src/fsharp/TcGlobals.fs

```diff
@@ -780,6 +784,14 @@ type public TcGlobals(compilingFslib: bool, ilg:ILGlobals, fslibCcu: CcuThunk, d
        | res -> res
     mkILCustomAttrs (attrs.AsList @ attribs)
 
+  let addValGeneratedAttrs (v: Val) m =
+    if not noDebugData then
+        v.SetAttribs ([
+           Attrib(v_CompilerGeneratedAttribute_tcr, ILAttrib ((mkILNonGenericCtorMethSpec (tref_CompilerGeneratedAttribute, [])).MethodRef), [], [], false, None, m)
```

Comment:
Unrelated comment: The `noDebugData` looks like legacy internal compiler flag thing.  We should probably remove it